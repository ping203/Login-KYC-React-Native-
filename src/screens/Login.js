import React, {Component} from 'react';
import {
  View,
  Text,
  Alert,
  Image,
  AsyncStorage,
  TouchableOpacity,
} from 'react-native';
import {TextField} from 'react-native-material-textfield';
import styles from '../components/styles';

export default class Login extends Component {
  componentDidMount = () => {
    AsyncStorage.getItem('username').then(value =>
      this.setState({username: value}),
    );
    AsyncStorage.getItem('password').then(value =>
      this.setState({password: value}),
    );
  };
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      inputUsername: '',
      inputPassword: '',
      hidePassword: true,
    };
  }
  setPasswordVisibility = () => {
    this.setState({hidePassword: !this.state.hidePassword});
  };
  myValidate = () => {
    const {inputUsername, inputPassword} = this.state;
    const myUsername = this.state.username;
    const myPassword = this.state.password;
    if (inputUsername == '' && inputPassword == '') {
      Alert.alert('Please fill the credentials');
    } else if (inputUsername != myUsername && inputPassword != myPassword) {
      Alert.alert('Account not found');
    } else if (inputUsername != '' && inputPassword != myPassword) {
      Alert.alert('Please fill the password correctly');
    } else if (inputUsername != myUsername && inputPassword != '') {
      Alert.alert('Please fill the username correctly');
    } else if (inputUsername == myUsername && inputPassword == myPassword) {
      this.props.navigation.navigate('TabNav');
    } else {
      Alert.alert('Data not found');
    }
  };
  render() {
    let {inputUsername} = this.state;
    let {inputPassword} = this.state;
    return (
      <View style={styles.loginContainer}>
        <Image
          source={require('../assets/images/avatar.jpeg')}
          style={{
            width: '50%',
            height: '50%',
            marginTop: 5,
            marginLeft: '30%',
            resizeMode: 'contain',
          }}
        />
        <View style={{padding: 15, marginBottom: -30, marginTop: -20}}>
          <TextField
            label="Username"
            value={inputUsername}
            autoCapitalize="none"
            onChangeText={inputUsername => this.setState({inputUsername})}
            tintColor="grey"
            activeLineWidth={1}
          />
        </View>

        <View style={{padding: 15}}>
          <TextField
            label="Password"
            value={this.state.inputPassword}
            onChangeText={inputPassword => this.setState({inputPassword})}
            tintColor="grey"
            activeLineWidth={1}
            secureTextEntry={this.state.hidePassword}
          />
          <TouchableOpacity
            activeOpacity={0.8}
            style={styles.btneye}
            onPress={this.setPasswordVisibility}>
            <Image
              style={styles.eye}
              source={
                this.state.hidePassword
                  ? require('../assets/images/Hide.png')
                  : require('../assets/images/Show.png')
              }
            />
          </TouchableOpacity>
        </View>
        <TouchableOpacity
          style={styles.buttonContainerStyle}
          activeOpacity={0.5}
          onPress={this.myValidate}>
          <View style={styles.loginButtonStyles}>
            <Text style={styles.loginTextStyle}>Login</Text>
          </View>
        </TouchableOpacity>
      </View>
    );
  }
}
