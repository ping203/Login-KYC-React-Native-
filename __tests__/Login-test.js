import React from 'react';
import Login from '../src/screens/Login';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
