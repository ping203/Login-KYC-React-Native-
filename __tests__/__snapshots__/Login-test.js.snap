// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 50,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/avatar.jpeg",
      }
    }
    style={
      Object {
        "height": "50%",
        "marginLeft": "30%",
        "marginTop": 5,
        "resizeMode": "contain",
        "width": "50%",
      }
    }
  />
  <View
    style={
      Object {
        "marginBottom": -30,
        "marginTop": -20,
        "padding": 15,
      }
    }
  >
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 16,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "bottom": 0,
                "left": -1.5,
                "position": "absolute",
                "right": -1.5,
                "top": -1.5,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "left": "-100%",
                "paddingLeft": "50%",
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "translateY": 20,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "200%",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 16,
                  "includeFontPadding": false,
                  "lineHeight": 16,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                }
              }
            >
              Username
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              disableFullscreenUI={true}
              editable={true}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onContentSizeChange={[Function]}
              onFocus={[Function]}
              rejectResponderTermination={true}
              selectionColor="grey"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "includeFontPadding": false,
                    "margin": 0,
                    "padding": 0,
                    "paddingTop": 0,
                    "textAlign": "left",
                    "textAlignVertical": "top",
                    "top": 2,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, .87)",
                    "fontSize": 16,
                    "height": 24,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "minHeight": 8,
            },
            Object {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 15,
      }
    }
  >
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "height": 64,
            "paddingBottom": 8,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 16,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.38)",
                "borderStyle": "solid",
                "borderWidth": 0.5,
                "bottom": 0,
                "left": -1.5,
                "position": "absolute",
                "right": -1.5,
                "top": -1.5,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "left": "-100%",
                "paddingLeft": "50%",
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "translateY": 20,
                  },
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": "200%",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 16,
                  "includeFontPadding": false,
                  "lineHeight": 16,
                  "textAlign": "left",
                  "textAlignVertical": "top",
                }
              }
            >
              Password
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="sentences"
              disableFullscreenUI={true}
              editable={true}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onContentSizeChange={[Function]}
              onFocus={[Function]}
              rejectResponderTermination={true}
              secureTextEntry={true}
              selectionColor="grey"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "includeFontPadding": false,
                    "margin": 0,
                    "padding": 0,
                    "paddingTop": 0,
                    "textAlign": "left",
                    "textAlignVertical": "top",
                    "top": 2,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, .87)",
                    "fontSize": 16,
                    "height": 24,
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "minHeight": 8,
            },
            Object {
              "paddingLeft": 0,
              "paddingRight": 0,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 80,
          "marginTop": 10,
          "opacity": 1,
          "padding": 2,
          "position": "absolute",
          "right": 30,
          "width": 35,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/images/Hide.png",
          }
        }
        style={
          Object {
            "height": "80%",
            "margin": 10,
            "resizeMode": "contain",
            "width": "80%",
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 50,
        "marginTop": 15,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00acee",
          "height": 50,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flexWrap": "wrap",
            "fontSize": 16,
            "padding": 15,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
